### PR TITLE
feat(ci): Merge agent 6 and 7 rules - part 1

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Remove distinction between agent6 and 7 rules. Building an agent alone is not used

### Motivation
Simplify the gitlab configuration

### Additional Notes
Test pipelines generated:
[standard deploy pipeline](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/31122580)
[Merge queue pipeline generation test](https://datadoghq.atlassian.net/wiki/x/5AGltQ)
[Branch comparing to itself for modifications on yaml files](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/31121318)
[integrations-core like pipeline](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/31122269)

### CODE REVIEW

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on

File: .gitlab/binary_build/cluster_agent_cloudfoundry.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 ---
 cluster_agent_cloudfoundry-build_amd64:
-  rules:
-    !reference [.on_a7]
```
Comment (KevinFairise2) [inline .gitlab/binary_build/cluster_agent_cloudfoundry.yml:4]: If you remove this rule these jobs will be created in the MQ, no?

**Review by chouquette — APPROVED** [review]
LGTM for BARX files

**Review by KevinFairise2 — COMMENTED** [review]
LGTM, when the issue with jobs created in the MQ is tackled. And if all the deleted rules are not used elsewhere

File: .gitlab/binary_build/cluster_agent_cloudfoundry.yml (hunk 1 above)
Comment (chouetz) [reply .gitlab/binary_build/cluster_agent_cloudfoundry.yml:4]: Good catch, the [test](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/31117488) was also relevant. Update on-going, thanks a lot

**Review by CelianR — APPROVED** [review]
Lgtm if the ci passes 👍

**Comment by chouetz** [comment]
> LGTM, when the issue with jobs created in the MQ is tackled. And if all the deleted rules are not used elsewhere

Added a link with the workflow generation test on [MQ](https://datadoghq.atlassian.net/wiki/x/5AGltQ)
Thanks again

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_KITCHEN_TESTS=false --variable RUN_E2E_TESTS=off --variable DEPLOY_AGENT=false --variable RELEASE_VERSION_6=nightly --variable RELEASE_VERSION_7=nightly-a7 --variable BUCKET_BRANCH=dev --variable INTEGRATIONS_CORE_VERSION=master

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on

**Review by clarkb7 — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Pull request merged**